internal/provider/cluster_resource.go

```diff
@@ -72,194 +76,110 @@ func (r clusterResourceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Di
-											"machine_type": {
-												Type:     types.StringType,
-												Optional: true,
-											},
-											"num_virtual_cpus": {
-												Type:     types.Int64Type,
-												Optional: true,
-											},
-										}),
-									},
-								}),
-							},
-						}),
+					"routing_id": {
+						Computed: true,
+						Type:     types.StringType,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
 					},
 				}),
 			},
-			"update_spec": {
+			"dedicated": {
 				Optional: true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.UseStateForUnknown(),
+				},
 				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-					"serverless": {
+					"storage_gib": {
+						Type:     types.Int64Type,
 						Optional: true,
-						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-							"spend_limit": {
-								Optional: true,
-								Type:     types.Int64Type,
-							},
-						}),
+						Computed: true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
 					},
-					"dedicated": {
+					"disk_iops": {
+						Type:     types.Int64Type,
 						Optional: true,
-						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-							"region_nodes": {
-								Type:     types.MapType{ElemType: types.Int64Type},
-								Optional: true,
-							},
-							"hardware": {
-								Optional: true,
-								Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-									"storage_gib": {
-										Type:     types.Int64Type,
-										Optional: true,
-									},
-									"disk_iops": {
-										Type:     types.Int64Type,
-										Optional: true,
-									},
-									"machine_spec": {
-										Optional: true,
-										Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-											"machine_type": {
-												Type:     types.StringType,
-												Optional: true,
-											},
-											"num_virtual_cpus": {
-												Type:     types.Int64Type,
-												Optional: true,
-											},
-										}),
-									},
-								}),
-							},
-						}),
+						Computed: true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
 					},
-				}),
-			},
-			"config": {
-				Computed: true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					tfsdk.UseStateForUnknown(),
-				},
-				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-					"serverless": {
+					"memory_gib": {
+						Computed: true,
+						Type:     types.Float64Type,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
+					},
+					"machine_type": {
+						Type:     types.StringType,
+						Optional: true,
 						Computed: true,
 						PlanModifiers: tfsdk.AttributePlanModifiers{
 							tfsdk.UseStateForUnknown(),
 						},
```

Comment:
When I tried to update the `machine_type` in dedicated cluster the TF provided gave inconsistent result. I have updated the `machine_type     = "n1-standard-4" -> "n1-standard-8"`

```
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cockroach_cluster.cockroach, provider "provider[\"registry.terraform.io/cockroachdb/cockroach\"]" produced an unexpected new
│ value: .dedicated.memory_gib: was cty.NumberIntVal(15), but now cty.NumberIntVal(30).
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cockroach_cluster.cockroach, provider "provider[\"registry.terraform.io/cockroachdb/cockroach\"]" produced an unexpected new
│ value: .dedicated.num_virtual_cpus: was cty.NumberIntVal(4), but now cty.NumberIntVal(8).
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```